File: Part1/Laba_3/Laba_3/RunF3.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Threading;

namespace Laba_3
{
    class RunF3
    {
        public Thread thread;
        int n;
        private int[] O;
        private int[] T;
        private int[][] MP;
        private int[][] MR;

        public RunF3(int n)
        {
            thread = new Thread(this.Run);
            this.n = n;
            thread.Priority = ThreadPriority.Lowest;
            Console.WriteLine("F3 started");
            thread.Start();
        }

        void Run()
        {
            T = new int[n];
            MP = new int[n][];
            MR = new int[n][];
            Data.GenerateVector(T);
            Data.GenerateMatrix(MP);
            Data.GenerateMatrix(MR);
            O = Data.F3(MP, MR, T);
            Console.WriteLine("F3 finished");
        }

        public void PrintResult()
        {
            Console.WriteLine("\nRResult of F3: O = MAX(MP*MR) * T:\n");
            Data.PrintVector(O);
        }
    }
}

File: Part1/Laba_2/src/main/java/RunF3.java
public class RunF3 implements Runnable {
    private int[] O;
    private int[] T;
    private int[][] MP;
    private int[][] MR;
    Thread t;
    int n;

    public RunF3(int n) {
        t = new Thread(this, "F1");
        this.n = n;
        System.out.println("F3 started");
        t.setPriority(1);
        t.start();
    }

    @Override
    public void run() {
        T = new int[n];
        MP = new int[n][n];
        MR = new int[n][n];
        Data.generateVector(T);
        Data.generateMatrix(MP);
        Data.generateMatrix(MR);
        O = Data.f3(MP, MR, T);
        System.out.println("F3 finished");
    }

    public void printResult() {
        System.out.println("\nResult of F3: O = MAX(MP*MR) * T:\n");
        Data.printVector(O);
    }
}
